### PR TITLE
Feature/shopping cart screen

### PR DESCRIPTION
**Pull request by Tiewhan**
<h1>Overview</h1>

- The shopping cart feature allows the user to add games to their shopping cart.
- An animation was added to the shopping cart button that triggers when the add to shopping cart button is pressed to notify the user that something happened
- User Can remove items from the shopping cart by swiping left

<h2>Notes</h2>
Various other small tweaks were made as they were discovered

- Colours were added to the asset registry to ease of use of app colour scheme
- Removed background colour code from GameTableCell as it because it was updated and the new look doesn't utilise it
- Removed "prepare for segue" code in game view as it was un needed (The base implementation for clicking a cell was inconsistent and a custom gesture recogniser is now used for each cell and it will trigger a separate action that moves to the game details screen)

<h2>Screenshots</h2>
<h3>Shopping Cart</h3>
Simulator Screen Shot - iPhone 11 Pro Max - 2020-04-20 at 08 44 27
<h3>Shopping Cart Remove From Cart</h3>
Simulator Screen Shot - iPhone 11 Pro Max - 2020-04-20 at 08 52 59

### CODE REVIEW

File: Feisty/GameView/GameViewComponents/GameTableViewCell.swift [hunk 1]
```diff
@@ -14,14 +14,9 @@ public class GameTableViewCell: UITableViewCell {
...
   @IBOutlet public weak var txtGamePrice: UILabel!
+  @IBOutlet weak var gameImage: UIImageView!
   public var cellTappedAction: (() -> Void)?
 
-  private lazy var backGroundView: UIView = {
-    let view = UIView()
-    view.backgroundColor = .darkGray
-    return view
-  }()
-
```
Comment (Tiewhan) [inline Feisty/GameView/GameViewComponents/GameTableViewCell.swift:24]: Dead code that set the background colour of the cell. No longer needed as the background colour of the cell is now set to clear.

File: Feisty/GameView/GameViewComponents/GamesViewController.swift [hunk 2]
```diff
@@ -71,33 +74,6 @@ class GamesViewController: UITableViewController {
...
-
-      let selectedGame = dataViewModel.getGameAt(at: indexPath.row)
-      gameDetailController.selectedGame = selectedGame
-
-    default:
-      fatalError("Unexpected Segue Identifier")
-    }
-
-  }
-
```
Comment (Tiewhan) [inline Feisty/GameView/GameViewComponents/GamesViewController.swift:100]: Dead code for "prepare for segue"

**Review by AkuaQ — APPROVED** [review]

File: Feisty/GameDetails/GameDetailsViewController.swift [hunk 3]
```diff
@@ -37,9 +40,64 @@ class GameDetailsViewController: UIViewController {
...
+  }
+  
+  private func setUpShoppingCartFloatingActionButton() {
+    
     shoppingCartFAB.backgroundColor = UIColor.clear
-    addToCartFAB.backgroundColor = UIColor.clear
     shoppingCartFAB.button.setImage(#imageLiteral(resourceName: "ShoppingCart"), for: .normal)
+    shoppingCartFAB.actionToPerform = {
+      
+      let storyboard = UIStoryboard(name: "Main", bundle: .main)
```
Comment (Zamo22) [inline Feisty/GameDetails/GameDetailsViewController.swift:54]: I would consider putting "Main" into a constant variable if you plan to use it in the future. Not required though

File: Feisty/GameDetails/GameDetailsViewController.swift [hunk 4]
```diff
@@ -37,9 +40,64 @@ class GameDetailsViewController: UIViewController {
...
+  private func setUpAddToCartFloatingActionButton() {
+    
+    addToCartFAB.backgroundColor = UIColor.clear
+    addToCartFAB.actionToPerform = { [weak self] in
+      
+      guard let self = self else {
+        return
+      }
+      
+      guard let game = self.selectedGame else {
```
Comment (Zamo22) [inline Feisty/GameDetails/GameDetailsViewController.swift]: These two guards can be one statement

File: Feisty/GameDetails/GameDetailsViewController.swift [hunk 5]
```diff
@@ -37,9 +40,64 @@ class GameDetailsViewController: UIViewController {
...
+      
+      self.shoppingCart?.shoppingList.append(game)
+      
+      self.shoppingCartRightConstraint.constant = 32
+      
+      let animations: () -> Void = { [weak self] in
+        self?.view.layoutIfNeeded()
+      }
+      
+      let completion: (Bool) -> Void = { [weak self] completed in
```
Comment (Zamo22) [inline Feisty/GameDetails/GameDetailsViewController.swift]: Bit confusing, why does this take a boolean but not actually use `completed` ?

Comment (Tiewhan) [reply Feisty/GameDetails/GameDetailsViewController.swift]: That's what the closure for animate requires. I did however use a "_" instead of completed.

**Review by Zamo22 — APPROVED** [review]

**Pull request merged**